### ISSUE TITLE
Search: Links marked as buttons without appropriate keyboard handling.

### ISSUE THREAD

**Issue by katiebethbrown** [Automattic/jetpack#23393]
### Impacted plugin
Jetpack

### Issue
In the search modal, numerous buttons are marked with role="button", causing them to be reported as buttons by screen readers. Buttons should be operable with either the space bar or the enter key, but since these are actually links, they are only operable using the enter key.

![](https://user-images.githubusercontent.com/1686451/129973707-21bf8076-0382-4fdb-91c7-24e24f9eb2da.png)

### Possible Fix
Should either remove the button role, so these are reported accurately, or add keyboard handlers.

### Other Information
This issue was surfaced to team 51 by an a11y contractor working on a partner site.
